lib/models/command.js

```diff
@@ -288,7 +288,11 @@ let Command = CoreObject.extend({
 
       if (this.works === 'insideProject') {
         if (!this.project.hasDependencies()) {
-          throw new SilentError('node_modules appears empty, you may need to run `npm install`');
+          let installInstuction = '`npm install`';
+          if (this.project.hasYarnLockFile()) {
```

Comment:
@rwjblue, we only use it in [/lib/commands/init.js:64](https://github.com/ember-cli/ember-cli/blob/bd2822868e3b9bf11cfa9f28b91cb4d021c6680b/lib/commands/init.js#L64) and [lib/tasks/npm-task.js:45](https://github.com/ember-cli/ember-cli/blob/bd2822868e3b9bf11cfa9f28b91cb4d021c6680b/lib/tasks/npm-task.js#L45)
I can take it out and check it directly if you like.
Than I just need another approach to test it, ATM, we overwrite the `hasYarnLockFile` function in the project class:
```js
command.project.hasYarnLockFile = function() { return true; };
```